addons/l10n_ro_edi_etransport/models/l10n_ro_edi_etransport_document.py

```diff
@@ -0,0 +1,563 @@
+
+    l10n_ro_edi_etransport_start_loc_value = fields.Char(compute='_compute_start_loc_value')
+    l10n_ro_edi_etransport_end_loc_value = fields.Char(compute='_compute_end_loc_value')
+
+    l10n_ro_edi_etransport_start_loc_type_index = fields.Integer(compute='_compute_start_loc_type_index')
+    l10n_ro_edi_etransport_end_loc_type_index = fields.Integer(compute='_compute_end_loc_type_index')
+
+    # Data fields for every location type
+    l10n_ro_edi_etransport_start_bcp = fields.Selection(selection=BORDER_CROSSING_POINTS, string="Border Crossing Point")
+    l10n_ro_edi_etransport_start_customs_office = fields.Many2one(comodel_name='l10n_ro.edi.etransport.customs', string="Customs Office")
+    l10n_ro_edi_etransport_start_state_id = fields.Many2one(comodel_name='res.country.state', string="State", domain="[('country_id.code', '=', 'RO')]")
+    l10n_ro_edi_etransport_start_city = fields.Char(string="City")
+    l10n_ro_edi_etransport_start_street = fields.Char(string="Street")
+    l10n_ro_edi_etransport_start_zip = fields.Char(string="Postal Code")
+    l10n_ro_edi_etransport_start_other_info = fields.Char(string="Other Info")
+
+    l10n_ro_edi_etransport_end_bcp = fields.Selection(selection=BORDER_CROSSING_POINTS, string="Border Crossing Point")
+    l10n_ro_edi_etransport_end_customs_office = fields.Many2one(comodel_name='l10n_ro.edi.etransport.customs', string="Customs Office")
+    l10n_ro_edi_etransport_end_state_id = fields.Many2one(comodel_name='res.country.state', string="State", domain="[('country_id.code', '=', 'RO')]")
+    l10n_ro_edi_etransport_end_city = fields.Char(string="City")
+    l10n_ro_edi_etransport_end_street = fields.Char(string="Street")
+    l10n_ro_edi_etransport_end_zip = fields.Char(string="Postal Code")
+    l10n_ro_edi_etransport_end_other_info = fields.Char(string="Other Info")
+
+    l10n_ro_edi_etransport_remarks = fields.Text(string="Remarks")
+
+    l10n_ro_edi_etransport_state = fields.Selection(selection=DOCUMENT_STATES, string="eTransport Status", default='draft', copy=False)
+    l10n_ro_edi_etransport_message = fields.Char(string="Message", copy=False)
+    l10n_ro_edi_etransport_uit = fields.Char(help="UIT of this eTransport document.", copy=False)
+    l10n_ro_edi_etransport_load_id = fields.Char(help="Id of this document used for interacting with the anaf api.", copy=False)
+
+    def _reset_location_type(self):
+        for doc in self:
+            doc.l10n_ro_edi_etransport_start_loc_type_1 = 'location'
+            doc.l10n_ro_edi_etransport_start_loc_type_2 = 'location'
+            doc.l10n_ro_edi_etransport_end_loc_type_1 = 'location'
+            doc.l10n_ro_edi_etransport_end_loc_type_2 = 'location'
+
+    def _reset_location_data(self, location: str):
+        for doc in self:
+            setattr(doc, f'l10n_ro_edi_etransport_{location}_bcp', None)
```

Comment:
`setattr` and `getattr` is a security risk. We should redesign this if possible.

Reply:
as far as i know this is allowed from a security perspective as long as the fields have a fixed prefix